google_symptoms/tests/test_date_utils.py

```diff
@@ -0,0 +1,92 @@
+            14,
+            True
+        )
+
+        expected = [datetime(2020, 1, 26),
+                    datetime(2021, 1, 5)]
+        assert set(output) == set(expected)
+
+    def test_generate_export_dates(self, params, logger, monkeypatch):
+        import covidcast
+        metadata_df = pd.read_csv(f"{TEST_DIR}/test_data/covid_metadata.csv")
+        monkeypatch.setattr(covidcast, "metadata", lambda: metadata_df)
+        num_export_days = generate_num_export_days(params, logger)
+
+        expected_num_export_days = params["indicator"]["num_export_days"]
+
+        assert num_export_days == expected_num_export_days
+
+    def test_generate_export_dates_normal(self, params_w_no_date, logger, monkeypatch):
+        import covidcast
+        metadata_df = pd.read_csv(f"{TEST_DIR}/test_data/covid_metadata.csv")
+        monkeypatch.setattr(covidcast, "metadata", lambda: metadata_df)
+        num_export_days = generate_num_export_days(params_w_no_date, logger)
+
+        max_expected_lag = lag_converter(params_w_no_date["validation"]["common"].get("max_expected_lag", {"all": 4}))
+        global_max_expected_lag = max(list(max_expected_lag.values()))
+        expected_num_export_days = params_w_no_date["validation"]["common"].get("span_length", 14) + global_max_expected_lag
+
+        assert num_export_days == expected_num_export_days
+
+    def test_generate_export_date_missing(self, params_w_no_date, logger, monkeypatch):
+        import covidcast
+        metadata_df = pd.read_csv(f"{TEST_DIR}/test_data/covid_metadata_missing.csv")
+        monkeypatch.setattr(covidcast, "metadata", lambda: metadata_df)
+        num_export_days = generate_num_export_days(params_w_no_date, logger)
+        expected_num_export_days = (date.today() - FULL_BKFILL_START_DATE.date()).days + 1
+        assert num_export_days == expected_num_export_days
+
+    def test_generate_patch_dates(self, params_w_patch, logger, monkeypatch):
+        import covidcast
+        metadata_df = pd.read_csv(f"{TEST_DIR}/test_data/covid_metadata_missing.csv")
+        monkeypatch.setattr(covidcast, "metadata", lambda: metadata_df)
+        max_expected_lag = lag_converter(params_w_patch["validation"]["common"].get("max_expected_lag", {"all": 4}))
+        global_max_expected_lag = max(list(max_expected_lag.values()))
+        expected_num_export_days = params_w_patch["validation"]["common"].get("span_length", 14) + global_max_expected_lag
+
+        issue_date = datetime.strptime(params_w_patch["patch"]["start_issue"], "%Y-%m-%d")
+        end_issue = datetime.strptime(params_w_patch["patch"]["end_issue"], "%Y-%m-%d")
+
+        patch_date_dict = generate_patch_dates(params_w_patch)
+
+        while issue_date <= end_issue:
+            expected_daterange = generate_query_dates(
+                FULL_BKFILL_START_DATE,
+                issue_date,
+                expected_num_export_days,
+                False
+            )
```

Comment:
Hm, this doesn't make sense to me. For an issue date `t`, we should be fetching data in the range [, `t` - `global_max_expected_lag`] (the data for days `t - 1`, `t - 2`, ..., `t - (global_max_expected_lag - 1)` wouldn't be available on date `t` because of the reporting lag

Reply:
I think [this PR](https://github.com/cmu-delphi/covidcast-indicators/pull/2007) fixes it.

Reply:
Yeah I was trying to replicate how the previous regular runs were going, which made things unnecessarily complicated; mentioned it in the PR, but adding it here for visibility